plasma_framework/test/src/utils/Merkle.test.js

```diff
@@ -5,38 +5,75 @@ const { expect } = require('chai');
 
 const { MerkleTree } = require('../../helpers/merkle.js');
 
+const Testlang = require('../../helpers/testlang.js');
+const config = require('../../../config.js');
+
 contract('Merkle', () => {
     const maxSize = 2 ** 16;
-    const leaves = [...Array(maxSize).keys()].map(index => `leaf ${index}`);
+    const DEPOSIT_VALUE = 1000000;
+    const OUTPUT_TYPE_PAYMENT = config.registerKeys.outputTypes.payment;
+    const MERKLE_TREE_DEPTH = 16;
+
+    const alicePrivateKey = '0x7151e5dab6f8e95b5436515b83f423c4df64fe4c6149f864daa209b26adb10ca';
+    const password = 'password1234';
 
-    before('setup merkle contract and tree value', async () => {
+    before('setup merkle contract', async () => {
         this.merkleContract = await Merkle.new();
-        this.merkleTree = new MerkleTree(leaves);
     });
 
     describe('checkMembership', () => {
+        const leaves = [...Array(maxSize).keys()].map(index => web3.utils.bytesToHex(`leaf ${index}`));
+
+        before('setup merkle contract and tree value', async () => {
+            this.merkleContract = await Merkle.new();
+            this.merkleTree = new MerkleTree(leaves);
+        });
+
         describe('Prove inclusion', () => {
             const testIndex = [0, 1, 2, 3, maxSize - 2, maxSize - 1];
 
             testIndex.forEach((leafIndex) => {
                 it(`should return true for index ${leafIndex}`, async () => {
-                    const leafData = web3.utils.sha3(leaves[leafIndex]);
                     const rootHash = this.merkleTree.root;
                     const proof = this.merkleTree.getInclusionProof(leaves[leafIndex]);
 
-                    const result = await this.merkleContract.checkMembership(leafData, leafIndex, rootHash, proof);
+                    const result = await this.merkleContract.checkMembership(
+                        leaves[leafIndex],
+                        leafIndex,
+                        rootHash,
+                        proof,
+                    );
                     expect(result).to.be.true;
                 });
             });
         });
 
         it('should return false when not able to prove included', async () => {
```

Comment:
It was asked by @thec00n a while ago, but if don't mind to put a small effort of adding a test, can you as well add one more in this file?

Something like the following:
```
it('should return false when leaf index mismatches the proof')
```

Basically we are missing tests when the index passed is wrong.